### PR TITLE
EWL-5799: search result molecule

### PR DESCRIPTION
**Pull request by labbydev**
## Ticket(s)
**Jira Ticket**
- [EWL-5799: Create Search Result Molecule](https://issues.ama-assn.org/browse/EWL-5799)
- [EWL-5800: Create Search Result as Best Bet variant](https://issues.ama-assn.org/browse/EWL-5800)

## Description
Create the Search Result Molecule and the "as Best Bet" variant.

## To Test
- [ ] `gulp serve`
- [ ] Navigate to http://localhost:3000/?p=molecules-search-result
- [ ] Observe that you see the subcategory with the all caps, purple style
- [ ] Observe that you see the title with the black link style
- [ ] Observe that you see the description of the content
- [ ] Navigate to http://localhost:3000/?p=molecules-search-result-as-best-bet
- [ ] Observe that you _do not_ see the subcategory
- [ ] Observe that you see the title and it's `$homepagePurple` and 26/31 
- [ ] Observe that you see the description
- [ ] Observe that you see an image on the left side
- [ ] Resize to mobile
- [ ] Observe that the image is not visible on mobile
- [ ] Did you test in IE 11?

## Visual Regressions

A report is available [here](http://htmlpreview.github.io/?https://github.com/AmericanMedicalAssociation/ama-style-guide-2/blob/visual-regression-testing-artifact/feature/EWL-5799-search-result/html_report/index.html).

## Relevant Screenshots/GIFs
screen shot 2018-08-22 at 5 40 41 pm
screen shot 2018-08-22 at 5 40 32 pm

## Remaining Tasks
N/A

## Additional Notes
N/A

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

File: styleguide/source/_patterns/02-molecules/search-result/search-result.twig [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+
+<div class="ama__search-result {{ classes|join(' ') }}">
+  <div class="ama__search-result__text">
+    {% if searchResult.category %}
+      {% include "@atoms/paragraph.twig" with { "paragraph" : searchResult.category } %}
+    {% endif %}
+
+    {% if link %}
+      {% set link = link|merge({class: 'ama__link--black', target: 'self'}) %}
```
Comment (JaeBea) [inline styleguide/source/_patterns/02-molecules/search-result/search-result.twig:17]: The class and target for the link should be defined in the json file.

**Review by JaeBea — CHANGES_REQUESTED** [review]
@labbydev ,
There are a few changes and questions I have for this work. I have made comments within the code changes when possible, the rest are described below.

**Changes**
- The best bets variant needs a markdown file.
- The scss for search results molecule and variants needs to be put in a folder called `search-results` to ensure the scss shows up in pattern info.

**Question**
- What is the ratios to be used on the best bets image? I cannot find documentation on it, but he image used in style guide will need to reflect that restriction.

File: styleguide/source/_patterns/02-molecules/search-result/search-result.twig (hunk 1 above)
Comment (labbydev) [reply styleguide/source/_patterns/02-molecules/search-result/search-result.twig:17]: The class should only be defined in the `json` when the class is possible to change. When things are not subject to be changed they should be set in `twig`.

**Comment by labbydev** [comment]
@JaeBea psudeo-patterns do not get markdown files. They are noted in the base pattern markdown only.

**Comment by labbydev** [comment]
Since there is no note dictating the size of the image. I have set it to use the same image that is used on the teaser within the subcategory index.

**Review by JaeBea — CHANGES_REQUESTED** [review]
@labbydev, thanks for updating the scss file structure. I hadn't realized the markdown wasn't needed for variants.

There is only one change request remaining. The image used on the subcategory index teaser is 566x370, but the best bets molecule uses 800x600. Could you update the image used in the best bets variant to match? Once that is done I can merge this in.

File: styleguide/source/_patterns/02-molecules/search-result/search-result~as-best-bet.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "searchResult": {
+    "image": {
+      "alt": "alt text",
+      "src": "https://ipsumimage.appspot.com/800x600x186?l=3:2|800x600&s=36",
```
Comment (labbydev) [inline styleguide/source/_patterns/02-molecules/search-result/search-result~as-best-bet.json]: @JaeBea The image that I'm using for the best bet is 800x600.

Comment (JaeBea) [reply styleguide/source/_patterns/02-molecules/search-result/search-result~as-best-bet.json]: @labbydev I am requesting that this is changed to match what is on the subcategory search area, 566x370. I have put a screenshot below.
image

Comment (labbydev) [reply styleguide/source/_patterns/02-molecules/search-result/search-result~as-best-bet.json]: Done

**Review by JaeBea — APPROVED** [review]
@labbydev this is now ready to merge. Thanks!

**Pull request merged**